### PR TITLE
#824 Fixed: behaviors are initiated with configuration data

### PR DESCRIPTION
**Pull request by belisoful**
with test unit.

### CODE REVIEW

File: framework/Util/IBaseBehavior.php [hunk 1]
```diff
@@ -22,6 +22,16 @@
  */
 interface IBaseBehavior
 {
+	/**
+	 * The array key for the $config data in instancing behaviors with init($config),
+	 */
+	public const CONFIG_KEY = "=config";
```
Comment (ctrlaltca) [inline framework/Util/IBaseBehavior.php:28]: Is "=config" intended to start with "="? To avoid people from setting it accidentally?

Comment (belisoful) [reply framework/Util/IBaseBehavior.php:28]: exactly.

Comment (belisoful) [reply framework/Util/IBaseBehavior.php:28]: '=' isn't a supported character for tags in xml.

**Review by ctrlaltca — APPROVED** [review]

**Comment by belisoful** [comment]
BTW, this is an important enhancement.  the new TAsset publishing code intends to do meta tag filtering and setting and setting new meta tags in jpegs would be be served not as just a class property but as a behavior xml tags on their own.

**Pull request merged**